Presentation/DualNavigationControllersSplitDelegate.swift

```diff
@@ -12,8 +12,25 @@ import Flow
 /// A split controller delegate (`UISplitViewControllerDelegate`), that manages navigation controllers for the master and detail view,
 /// as well as moving view controllers between these while expanding or collapsing.
 open class DualNavigationControllersSplitDelegate: NSObject, UISplitViewControllerDelegate {
-    private let isCollapsedProperty = ReadWriteSignal(true)
+
+    @available(*, deprecated, message: "use `init(collapsedState:)` instead")
+    public convenience override init() {
+        self.init(collapsedState: Future(true))
+    }
+
+    /// Creates a delegate that will manage a split view controller with the specified initial collapsed state
+    /// - Parameter collapsedState: Use to report the initial `isCollapsed` state of the split view controller.
+    /// Note that its value can be unreliable until it is rendered on the screen.
+    public init(collapsedState: Future<Bool>) {
+        super.init()
+        initialCollapsedStateDisposable += collapsedState.onValue { [weak self] in
```

Comment:
Is `initialCollapsedStateDisposable` necessary, as we hold self weakly anyway?

Reply:
I use the bag only to remove the subscription earlier if one of the other 2 delegate methods setting the state get triggered before the future completes. However that might not be very realistic case, what do you think?